### PR TITLE
File prefix format tests

### PR DESCRIPTION
**Pull request by awalter-bnl**
This adds a test of the file_prefix templating to `suitcase.tiff_stack`

### CODE REVIEW

File: suitcase/tiff_stack/tests/tests.py [hunk 1]
```diff
@@ -24,3 +24,27 @@ def test_export(tmp_path, example_data):
...
+        detectors and event_types. see `suitcase.utils.conftest` for more info.
+
+    '''
+    collector = example_data()
+    file_prefix = file_prefix_list()
+    artifacts = export(collector, tmp_path, file_prefix=file_prefix)
+
+    for name, doc in collector:
+        if name == 'start':
+            templated_file_prefix = file_prefix.format(**doc).partition('-')[0]
```
Comment (mrakitin) [inline suitcase/tiff_stack/tests/tests.py]: Maybe add a `break` there once the condition is satisfied?

File: suitcase/tiff_stack/tests/tests.py [hunk 2]
```diff
@@ -24,3 +24,27 @@ def test_export(tmp_path, example_data):
...
+
+
+def test_file_prefix_formatting(file_prefix_list, example_data, tmp_path):
+    ''' runs a test of the file_prefix formatting.
+
+    ..note::
+
+        Due to the `file_prefix_list` and `example_data` `pytest.fixture`'s
+        this will run multiple tests each with a range of file_prefixes,
+        detectors and event_types. see `suitcase.utils.conftest` for more info.
```
Comment (mrakitin) [inline suitcase/tiff_stack/tests/tests.py]:
```suggestion
        detectors and event_types. See `suitcase.utils.conftest` for more info.
```

File: suitcase/tiff_stack/tests/tests.py [hunk 3]
```diff
@@ -24,3 +24,27 @@ def test_export(tmp_path, example_data):
         actual = tifffile.imread(str(filename))
         streamname = os.path.basename(filename).split('-')[0]
         assert_array_equal(actual, expected[streamname])
+
+
+def test_file_prefix_formatting(file_prefix_list, example_data, tmp_path):
+    ''' runs a test of the file_prefix formatting.
```
Comment (mrakitin) [inline suitcase/tiff_stack/tests/tests.py]:
```suggestion
    '''Runs a test of the ``file_prefix`` formatting.
```

File: suitcase/tiff_stack/tests/tests.py [hunk 4]
```diff
@@ -6,13 +6,13 @@
...
 
 def test_export(tmp_path, example_data):
-    ''' runs a test using the plan that is passed through to it
+    ''' runs a test using the plan that is passed through to it.
 
     ..note::
 
-        Due to the `events_data` `pytest.fixture` this will run multiple tests
+        Due to the `example_data` `pytest.fixture` this will run multiple tests
         each with a range of detectors and a range of event_types. see
```
Comment (mrakitin) [inline suitcase/tiff_stack/tests/tests.py]:
```suggestion
        each with a range of detectors and a range of event_types. See
```

File: suitcase/tiff_stack/tests/tests.py [hunk 5]
```diff
@@ -6,13 +6,13 @@
 
 
 def test_export(tmp_path, example_data):
-    ''' runs a test using the plan that is passed through to it
+    ''' runs a test using the plan that is passed through to it.
```
Comment (mrakitin) [inline suitcase/tiff_stack/tests/tests.py]:
```suggestion
    '''Runs a test using the plan that is passed through to it.
```

File: suitcase/tiff_stack/tests/tests.py (hunk 1 above)
Comment (awalter-bnl) [reply suitcase/tiff_stack/tests/tests.py]: fixed

**Comment by awalter-bnl** [comment]
thanks for the quick review @mrakitin much appreciated.

**Comment by awalter-bnl** [comment]
I am not sure at all why adding one line (a break to the if statement on line 46) suddenly caused codecov to complain about the 2 lines above. I am open to suggestions on fixing this (short of removing the break, which I think is a good suggestion made by @mrakitin ).

**Comment by danielballan** [comment]
I have verified with a print statement that those lines are being hitting, so I think have to chalk this up to a bug in `coverage` itself. Not worth holding up this PR over, anyway.

**Pull request merged**